libraries/AP_InertialSensor/AP_InertialSensor.cpp

```diff
@@ -1099,18 +1099,49 @@ void AP_InertialSensor::periodic()
   accel_sample must be correctly scaled, offset and oriented for the
   board
 */
-bool AP_InertialSensor::_calculate_trim(const Vector3f &accel_sample, float& trim_roll, float& trim_pitch)
+bool AP_InertialSensor::_calculate_trim(const Vector3f &accel_sample, Vector3f &trim)
 {
-    trim_pitch = atan2f(accel_sample.x, norm(accel_sample.y, accel_sample.z));
-    trim_roll = atan2f(-accel_sample.y, -accel_sample.z);
-    if (fabsf(trim_roll) > radians(10) ||
-        fabsf(trim_pitch) > radians(10)) {
+    // allow multiple rotations, this allows us to cope with tailsitters
+    const enum Rotation rotations[] = {ROTATION_NONE,
+#ifndef HAL_BUILD_AP_PERIPH
+                                       AP::ahrs().get_view_rotation()
+#endif
+    };
+    bool good_trim = false;
+    for (auto r : rotations) {
+        trim.zero();
+
+        switch (r) {
+        case ROTATION_NONE:
+            trim.y = atan2f(accel_sample.x, norm(accel_sample.y, accel_sample.z));
+            trim.x = atan2f(-accel_sample.y, -accel_sample.z);
+            break;
+
+        case ROTATION_PITCH_90: {
```

Comment:
Does this keep the trim.x ?

Reply:
Can we keep x if it is just a level cal and wipe if it is a full accel cal?

Reply:
Also if we can keep the z for when doing a  level cal only then it means a full accel cal with nose up followed by a level cal with nose level (FW attitude) would give us FW roll.

Reply:
done